src/shared/lib/markDownRules.tsx

```diff
@@ -547,7 +575,52 @@ const getContainerAlignTag = (parents: ASTNode[]): AlignmentTag | undefined => {
 };
 
 export const preprocessImageLinks = (content: string) => {
-  return content?.replace(/(!\[.*\]\s*\(.*?) =\d*x\d*(\))/g, '$1$2');
+  const regexSize = /\s*=\s*(\d+)x(\d+)/g;
+  const regexImage = /!\[.*?\]\((.*?)\)/g;
+
+  const getImageSize = (url: string) => {
+    return new Promise<{ width: number; height: number }>((resolve, reject) => {
+      Image.getSize(
+        url,
+        (width, height) => resolve({ width, height }),
+        error => reject(error),
+      );
+    });
+  };
```

Comment:
Please move the `getImageSize` function outside of the `preprocessImageLinks` scope and make it exportable. So that we could cover it with tests later (or now?).

Also, if the `error` is not used anywhere in the code it might be a good idea to just resolve the promise with `null`. Otherwise, the code that utilizes `getImageSize` should handle the error, e.g. put it in the logs